crates/core/src/subscription/subscription.rs

```diff
@@ -342,47 +353,64 @@ impl QuerySet {
     /// This is a *major* difference with normal query execution, where is expected to return the full result set for each query.
     #[tracing::instrument(skip_all)]
     pub fn eval(&self, db: &RelationalDB, tx: &Tx, auth: AuthCtx) -> Result<DatabaseUpdate, DBError> {
-        let mut database_update: DatabaseUpdate = DatabaseUpdate { tables: vec![] };
-        let mut table_ops = HashMap::new();
-        let mut seen = HashSet::new();
-
-        for SupportedQuery { expr, .. } in self {
-            if let Some(t) = expr.source.get_db_table() {
+        // evaluate each of the queries in this QuerySet in parallel
+        let ops = self
+            .par_iter()
+            .filter_map(|SupportedQuery { expr, .. }| expr.source.get_db_table().map(|t| (expr, t)))
+            .try_fold(Vec::new, |mut v, (expr, t)| {
+                #[cfg(feature = "metrics")]
                 let start = Instant::now();
-                // Get the TableOps for this table
-                let (_, table_row_operations) = table_ops
-                    .entry(t.table_id)
-                    .or_insert_with(|| (t.head.table_name.clone(), vec![]));
-                for table in run_query(&ExecutionContext::subscribe(db.address()), db, tx, expr, auth)? {
-                    for row in table.data {
-                        let row_pk = RelationalDB::pk_for_row(&row);
-
-                        //Skip rows that are already resolved in a previous subscription...
-                        if seen.contains(&(t.table_id, row_pk)) {
-                            continue;
-                        }
-                        seen.insert((t.table_id, row_pk));
-
-                        let row_pk = row_pk.to_bytes();
-                        table_row_operations.push(TableOp {
-                            op_type: 1, // Insert
-                            row_pk,
-                            row,
-                        });
-                    }
-                }
+                let tables = run_query(&ExecutionContext::subscribe(db.address()), db, tx, expr, auth)?;
+                v.push((t, tables));
                 #[cfg(feature = "metrics")]
                 record_query_duration_metrics(WorkloadType::Subscribe, &db.address(), start);
```

Comment:
Feel free to remove this. We're already collecting sufficient query metrics without it.